### PR TITLE
solution

### PR DESCRIPTION
**Pull request by ihorandrianov**
[DEMO LINK](https://ihorandrianov.github.io/react_keyboard/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,47 @@
...
+      this.setState({
+        key: e.key,
+      });
+    });
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', (e) => {
+      this.setState({
+        key: e.key,
```
Comment (MaxManis) [inline src/App.tsx:23]: this way of removing of event listener wont work, because you passed a completely new function)

File: src/App.tsx [hunk 2]
```diff
@@ -1,7 +1,47 @@
...
+            ? (
+              <div>
+                {`The last pressed key is ${this.state.key}`}
+              </div>
+            )
+            : (
+              <div>
+                Nothing was pressed yet
+              </div>
+            )
```
Comment (MaxManis) [inline src/App.tsx:42]: you also could use this ternaty statement inside of your div since its doesnt change)

File: src/App.tsx [hunk 3]
```diff
@@ -1,7 +1,45 @@
...
+        }
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:41]: Div's are same, just choose needed string by ternary operator

File: src/App.tsx [hunk 4]
```diff
@@ -1,7 +1,45 @@
...
+    this.setState({
+      key: e.key,
+    });
+  };
+
+  render() {
+    return (
+      <>
+        {
+          this.state.key
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:30]: Would be nice to use destructuring for state

**Review by OleksiiBlezniuk — CHANGES_REQUESTED** [review]

**Review by silchyn — APPROVED** [review]
Cool!

**Pull request closed without merging**